build.sbt

```diff
@@ -140,31 +153,53 @@ lazy val docs = project.dependsOn(core, generic, parser, optics)
   )
   .enablePlugins(MicrositesPlugin)
 
-lazy val aggregatedProjects: Seq[ProjectReference] = Seq[ProjectReference](
-  numbers, numbersJS,
-  core, coreJS,
-  generic, genericJS,
-  genericExtras, genericExtrasJS,
-  shapes, shapesJS,
-  literal, literalJS,
-  refined, refinedJS,
-  parser, parserJS,
-  scodec, scodecJS,
-  testing, testingJS,
-  tests, testsJS,
+lazy val circeCrossModules = Seq[CrossProject](
+  numbersBase,
+  coreBase,
+  genericBase,
+  genericExtrasBase,
+  shapesBase,
+  literalBase,
+  opticsBase,
+  refinedBase,
+  parserBase,
+  scodecBase,
+  testingBase,
+  testsBase
+)
+
+lazy val circeJsModules = Seq[Project](scalajs)
+
+lazy val circeJvmModules = Seq[Project](
   jawn,
   jackson,
-  optics, opticsJS,
-  scalajs,
-  streaming,
+  java8,
+  streaming
+)
+
+lazy val circeUtilModules = Seq[Project](
   docs,
   spray,
   hygiene,
   benchmark
-) ++ (
-  if (sys.props("java.specification.version") == "1.8") Seq[ProjectReference](java8) else Nil
 )
 
+def jvmVersionFilter: Set[Project] = sys.props("java.specification.version") match {
+  case "1.8"         => Set.empty
+  case "1.6" | "1.7" => Set(java8, optics)
```

Comment:
The latest version of Monocle supports Java 7 for the Scala 2.10 build, but not for 2.11 or 2.12.